python/beeai_framework/backend/errors.py

```diff
@@ -40,6 +43,19 @@ def __init__(
     ) -> None:
         super().__init__(message, is_fatal=True, is_retryable=False, cause=cause, context=context)
 
+    @classmethod
+    def ensure(
+        cls,
+        error: Exception,
+        *,
+        message: str | None = None,
+        context: dict[str, Any] | None = None,
+        model: Optional["ChatModel"] = None,
+    ) -> "FrameworkError":
+        model_context = {"provider": model.provider_id, "model_id": model.model_id} if model is not None else {}
+        context.update(model_context) if context is not None else None
+        return super().ensure(error, message=message, context=context)
```

Comment:
We should be using a separate class, as in TS.

`ChatModelError`

Reply:
This is the `ChatModelError` class though? Could you clarify for me?

Reply:
Do you mean i shouldn't call `super()` but instead init a `ChatModelError`?

Reply:
Sorry; I was looking wrong. All good except the `context` mutation.

Reply:
> All good except the `context` mutation.

Already fixed :)